bare_metal_billing/main.py

```diff
@@ -0,0 +1,166 @@
+        default=default_end_argument(),
+        type=parse_time_argument,
+        help=(
+            "End of the invoicing period. (YYYY-MM-DD)."
+            " Not inclusive. Defaults to today."
+        ),
+    )
+    parser.add_argument(
+        "--invoice-month",
+        default=default_start_argument().strftime("%Y-%m"),
+        help=(
+            "Use the first column for Invoice Month, rather than Interval."
+            " Defaults to month of start. (YYYY-MM)."
+        ),
+    )
+    parser.add_argument(
+        "bm_usage_file",
+        help="One or more CSV files that need to be processed",
+    )
+    parser.add_argument(
+        "--rate-fc430-su", default=0, type=Decimal, help="Rate of FC430 SU/hr"
+    )
+    parser.add_argument(
+        "--rate-fc830-su", default=0, type=Decimal, help="Rate of FC830 SU/hr"
+    )
+    parser.add_argument(
+        "--rate-gpu-a100sxm4-su",
+        default=0,
+        type=Decimal,
+        help="Rate of GPU A100 SXM4 SU/hr",
+    )
+    parser.add_argument(
+        "--rate-gpu-h100-su", default=0, type=Decimal, help="Rate of GPU H100 SU/hr"
+    )
+    parser.add_argument(
+        "--output-file",
+        default="bm_invoices.csv",
+        help="Output path for invoice in CSV format.",
+    )
+    parser.add_argument(
+        "--use-nerc-rates",
+        action="store_true",
+        help=(
+            "Set to use usage rates and su definitions from nerc-rates repo"
+            "instead of cli arguements. If not set, su defintions will default"
+            "to 1 for each SU's resources"
+        ),
+    )
+
+    args = parser.parse_args()
+
+    logger.info(f"Processing invoices for month {args.invoice_month}.")
+    logger.info(f"Interval for processing {args.start} - {args.end}.")
+    logger.info(f"Invoice file will be saved to {args.output_file}.")
+
+    su_info_dict = {}
+    su_rates_dict = {}
+    if args.use_nerc_rates:
+
+        def get_decimal_rate(rate_name):
+            return Decimal(nerc_repo_rates.get_value_at(rate_name, args.invoice_month))
+
+        nerc_repo_rates = load_from_url()
+        for su_name in billing.BM_SU_LIST:
+            su_rates_dict[su_name] = get_decimal_rate(f"{su_name} SU Rate")
+            su_info_dict[su_name] = {}
+            for su_resource in billing.SU_RESOURCE_LIST:
+                su_info_dict[su_name][su_resource] = get_decimal_rate(
+                    f"{su_resource} in {su_name} SU"
+                )
+    else:
+        su_rates_dict = {
+            "BM FC430": args.rate_fc430_su,
+            "BM FC830": args.rate_fc830_su,
+            "BM GPUA100SXM4": args.rate_gpu_a100sxm4_su,
+            "BM GPUH100": args.rate_gpu_h100_su,
+        }
+        for su_name in billing.BM_SU_LIST:
+            su_info_dict[su_name] = {}
+            for su_resource in billing.SU_RESOURCE_LIST:
+                su_info_dict[su_name][su_resource] = Decimal(1)
+
+    # TODO What are reasonable values for UNKNOWN SUs?
```

Comment:
@knikolla @naved001 @joachimweyl

Reply:
If I am unable to determine the SU then for an unknown SU it's 0. My rationale is that we must be sure what the SU is if we want to charge someone.

Ultimately, in the final report we will should see how many hours of unknown SU a project consumed and investigate the cause of it.